lib/Doctrine/Record.php

```diff
@@ -887,18 +887,33 @@ public function getSerializeVars()
      */
     public function serialize()
     {
+        return serialize($this->__serialize());
+    }
+
+    /**
+     * As of PHP 8.1.0, a class which implements Serializable without also implementing __serialize() and __unserialize() will generate a deprecation warning.
+     * @see https://php.watch/versions/8.1/serializable-deprecated
+     *
+     * @return array
+     */
+    public function __serialize()
+    {
+        $manager    = Doctrine_Manager::getInstance();
+        $connection = $manager->getConnectionForComponent(get_class($this));
+
+        $this->_table = $connection->getTable(get_class($this));
```

Comment:
@marcing this is new. Is setting `$this->_table` really necessary in this method now?